langchain4j/src/main/java/dev/langchain4j/data/document/DocumentLoaderUtils.java

```diff
@@ -38,18 +36,16 @@ static DocumentType detectDocumentType(String pathToFile) {
             return PDF;
         }
 
-        if (pathToFile.endsWith(".ppt")
-                || pathToFile.endsWith(".pptx")
-                || pathToFile.endsWith(".doc")
-                || pathToFile.endsWith(".docx")
-                || pathToFile.endsWith(".xls")
-                || pathToFile.endsWith(".xlsx")) {
-            return Arrays.stream(pathToFile.toUpperCase().split("\\."))
-                    .sorted(Comparator.reverseOrder())
-                    .findFirst()
-                    .map((extension) -> extension.endsWith("X") ? extension.replaceFirst(".$", "") : extension)
-                    .map(DocumentType::valueOf)
-                    .get();
+        if (pathToFile.endsWith(".doc") || pathToFile.endsWith(".docx")) {
+            return DOC;
+        }
+
+        if (pathToFile.endsWith(".xls") || pathToFile.endsWith(".xlsx")) {
+            return XLS;
+        }
+
+        if (pathToFile.endsWith(".ppt") || pathToFile.endsWith(".pptx")) {
+            return PPT;
```

Comment:
How about moving the available extensions and the method for checking that into DocumentType? 
If there are too many types, it might become inconvenient.

Reply:
@oognuyh good idea! will do